pkg/k8s/node.go

```diff
@@ -237,15 +249,68 @@ func setNodeNetworkUnavailableFalse(c kubernetes.Interface, nodeName string) err
 	return err
 }
 
+// removeNodeTaint removes the AgentNotReadyNodeTaint allowing for pods to be
+// scheduled once Cilium is setup. Mostly used in cloud providers to prevent
+// existing CNI plugins from managing pods.
+func removeNodeTaint(ctx context.Context, c kubernetes.Interface, nodeGetter nodeGetter, nodeName string) error {
+	k8sNode, err := nodeGetter.GetK8sNode(ctx, nodeName)
+	if err != nil {
+		return err
+	}
+
+	var taintFound bool
+
+	var taints []corev1.Taint
```

Comment:
```suggestion
	taints := make([]corev1.Taint, 0, len(k8sNode.Spec.Taints))
```

Reply:
taints won't have the same size so there's no point in allocating more than we need.